ably-ios/ARTRest.m

```diff
@@ -200,10 +207,20 @@ - (void)time:(void(^)(NSDate *time, NSError *error))callback {
     return nil;
 }
 
-- (void)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback {
-    NSParameterAssert(query.limit < 1000);
-    NSParameterAssert([query.start compare:query.end] != NSOrderedDescending);
-    
+- (BOOL)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback error:(NSError **)errorPtr {
+    if (query.limit > 1000 && errorPtr) {
```

Comment:
I get that if errorPtr is nil then the caller ignores the error, but I don't think that we should just continue when the preconditions are wrong. We should return early to avoid an inconsistent state that could even crash the program.